hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -279,7 +279,7 @@ private void saveInternalSchema(HoodieTable table, String instantTime, HoodieCom
     FileBasedInternalSchemaStorageManager schemasManager = new FileBasedInternalSchemaStorageManager(table.getMetaClient());
     if (!historySchemaStr.isEmpty() || Boolean.parseBoolean(config.getString(HoodieCommonConfig.RECONCILE_SCHEMA.key()))) {
       InternalSchema internalSchema;
-      Schema avroSchema = HoodieAvroUtils.createHoodieWriteSchema(new Schema.Parser().parse(config.getSchema()));
+      Schema avroSchema = HoodieAvroUtils.addMetadataFields(new Schema.Parser().parse(config.getSchema()), config.allowOperationMetadataField());
       if (historySchemaStr.isEmpty()) {
```

Comment:
Why not add a new method: HoodieAvroUtils#createHoodieWriteSchema(HoodieWriteConfig)

Reply:
1) Because `HoodieAvroUtils` from `hudi-common` while `HoodieWriteConfig` from `hudi-client-common` which depends on the former
2) And workaround (1) using `HoodieConfig` instead of `HoodieWriteConfig` does not make sense because there is no `getSchema()` in `HoodieConfig`. There is no even `AVRO_SCHEMA_STRING` to `HoodieConfig.getString(AVRO_SCHEMA_STRING)`. I think raw string `hoodie.avro.schema` is an inappropriate approach

Reply:
Yes it does. Fine I'll replace usage of `HoodieAvroUtils.addMetadataFields(Schema, boolean)`
with `HoodieAvroUtils.createHoodieWriteSchema(String, boolean)`